### PR TITLE
Add CNN Production model

### LINKED ISSUE #63: Add production CNN model
- Add model code
- load weights from Hugging Face - link to come
- test that it works?

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Add loading production CNN model

Fixes #63 

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_forecast/models/cnn/model.py [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        use_hf: bool = False,
+    ):
+        """
+        Load model weights
+        """
+
+        if use_hf:
+            return Model.from_pretrained("openclimatefix/nowcasting_cnn")
+
+        if remote_filename is None:
```
Comment (peterdudfield) [inline nowcasting_forecast/models/cnn/model.py:327]: im tempted to get rid of this, mauybe keep the option to load from local_filename, but the hugging_face method should be the one to use

File: nowcasting_forecast/models/hub.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+MODEL_CARD_MARKDOWN = """---
+license: mit
+tags:
+- nowcasting
+- forecasting
+- timeseries
+- remote-sensing
+---
+
+# {model_name}
```
Comment (peterdudfield) [inline nowcasting_forecast/models/hub.py:34]: do we need to fill this in? I can

File: requirements.txt [hunk 3]
```diff
@@ -3,11 +3,13 @@ sqlalchemy
 psycopg2-binary
 numpy
 click
-nowcasting_dataset==3.3.4
+nowcasting_dataset==3.3.9
+nowcasting_dataloader
```
Comment (peterdudfield) [inline requirements.txt:7]: we should pin the dataloader

**Comment by peterdudfield** [comment]
> @peterdudfield The only failing tests has to do with the NetCDFDataset, not the model.

Thanks @jacobbieker for all this. Ill have a look at the failing test. There's a few comments but i'm happy to help fix them

File: nowcasting_forecast/models/hub.py (hunk 2 above)
Comment (jacobbieker) [reply nowcasting_forecast/models/hub.py:34]: Probably, it is what shows up on the model card in HF. So more of a description is probably helpful.

File: nowcasting_forecast/models/cnn/model.py (hunk 1 above)
Comment (jacobbieker) [reply nowcasting_forecast/models/cnn/model.py:327]: Okay, yeah, sounds good!

**Comment by jacobbieker** [comment]
> > @peterdudfield The only failing tests has to do with the NetCDFDataset, not the model.
> 
> Thanks @jacobbieker for all this. Ill have a look at the failing test. There's a few comments but i'm happy to help fix them

Thanks! I've addressed most of them now. But if you could write the description of the model and fill out the model card that would be great!

File: nowcasting_forecast/models/cnn/cnn.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                    gsp_id=gsp_id,
+                )
+            )
+
+    forecasts = pd.DataFrame(forecasts)
+
+    return forecasts
+
+
+def run_cnn_batch(batch: Batch, model) -> xr.DataArray:
```
Comment (peterdudfield) [inline nowcasting_forecast/models/cnn/cnn.py:85]: delete this function, we dont need this. This is copeid over from nwp tsimple

**Pull request merged**